lib/ruby_lsp/requests/on_type_formatting.rb

```diff
@@ -171,6 +175,25 @@ def find_indentation(line)
 
         count
       end
+
+      sig { void }
+      def auto_indent_after_end_keyword
+        start_line_index = (@position[:line] - 2).downto(0).find do |i|
+          END_REGEXES.any? { |regex| regex.match?(@lines[i]) }
+        end
+
+        return unless start_line_index
+
+        start_line_indentation = find_indentation(@lines[start_line_index])
+
+        (start_line_index + 1...@position[:line] - 1).each do |i|
+          add_edit_with_text("  ", { line: i, character: 0 })
+        end
+
+        move_cursor_to(@position[:line] - 1, 3)
+
+        add_edit_with_text("\n" + (" " * (start_line_indentation + 2)), @position)
```

Comment:
If I understand correctly, the code above that does `(start_line_index + 1...@position[:line] - 1).each` is the part pushing the indentation.

What's this line pushing forward? The next line after the `end`?

Reply:
Not only the next line after the end but all lines between "end" and current position